### ISSUE TITLE
Missing warning with default expression converted to tuple

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#35815]
The following program throws a `NullReferenceException` but there are no diagnostics at compile-time.

``` c#
#nullable enable

namespace ConsoleApp6
{
    class Program
    {
        static void Main(string[] args)
        {
            M(default);
        }

        static void M((string, int) p)
        {
            p.Item1.ToString();
        }
    }
}
```

**Comment by RikkiGibson** [comment]
This might actually be a symptom of #35815

**Comment by jcouv** [comment]
I think this is an expected consequence of the "default loophole" which LDM acknowledged: https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-03-04.md#the-default-loophole

**Comment by RikkiGibson** [comment]
Just to make sure I fully understand: when we pass a struct with non-nullable reference type fields off to a method, we don't warn if the flow state of some of those non-nullable fields are maybe-null?

**Comment by cston** [comment]
@RikkiGibson, that is correct. The opportunity to warn is when the `default` value of the `struct` is created.

**Comment by CyrusNajmabadi** [comment]
@jcouv  If this is working as "by design", could we close this out?
